aggregator/src/aggregator.rs

```diff
@@ -1771,345 +1779,329 @@ impl VdafOps {
             let aggregation_job_id = *aggregation_job_id;
             let verify_key = *verify_key;
             let agg_param = Arc::clone(&agg_param);
-            move || -> Result<Vec<ReportShareData<SEED_SIZE, A>>, Error> {
-                let span = info_span!(
-                    parent: parent_span,
-                    "handle_aggregate_init_generic threadpool task"
-                );
-                let _entered = span.enter();
-
-                // Decrypt shares & prepare initialization states. (§4.4.4.1)
-                let mut report_share_data = Vec::new();
-                for (ord, prepare_init) in req.prepare_inits().iter().enumerate() {
-                    // If decryption fails, then the aggregator MUST fail with error `hpke-decrypt-error`. (§4.4.2.2)
-                    let global_hpke_keypair = global_hpke_keypairs.keypair(
-                        prepare_init
-                            .report_share()
-                            .encrypted_input_share()
-                            .config_id(),
-                    );
 
-                    let task_hpke_keypair = task.hpke_keys().get(
-                        prepare_init
-                            .report_share()
-                            .encrypted_input_share()
-                            .config_id(),
-                    );
+            move || {
+                let span = info_span!(parent: parent_span, "handle_aggregate_init_generic threadpool task");
+                let _entered = span.enter();
```

Comment:
This is going to happen on the Tokio blocking thread, but we don't do much here, just kick off Rayon tasks and wait for them. The relevant spans and events will be happening on Rayon threads. Thus, we should pass the span in a tuple along with the `sender` to be cloned onto Rayon threads, and enter it at the top of the closure passed to `try_for_each_with()`. The impact of this issue is that the "VDAF preparation (helper initialization)" span below would be a root span, and various events would have no parent span.